### DISCUSSION TITLE
How to run a command after another one completed?

### DISCUSSION THREAD

**Discussion by ibizaman** [magit/magit#5007] [Support] [answered]
My workflow when working on a feature branch is to:

1. Run git fetch
2. Reset main branch to origin/master
3. Rebase feature branch on top of main

I’d like to automate this with one command. I came pretty close to succeeding but my issue is the magit git fetch is asynchronous. How can I make it so step 2 and 3 only start when step 1 has finished?

**Comment by kyleam** [comment, marked as answer]
You can set a sentinel for the async process.  Grepping `set-process-sentinel` in the code base will give a bunch of examples.  Here's one:

https://github.com/magit/magit/blob/141dd46798e5cae57617e941418ebbb3a2172f5e/lisp/magit-branch.el#L290-L296

**Reply by ibizaman** [reply]
Thank you!

**Reply by kyleam** [reply]
Sure. 

Another thing worth considering: it may be cleaner to just fetch synchronously with `magit-run-git` rather than trying to use `magit-fetch`/`magit-git-fetch`.
